pkg/registry/core/pod/strategy.go

```diff
@@ -240,18 +240,22 @@ func ToSelectableFields(pod *api.Pod) fields.Set {
 	// amount of allocations needed to create the fields.Set. If you add any
 	// field here or the number of object-meta related fields changes, this should
 	// be adjusted.
-	podSpecificFieldsSet := make(fields.Set, 9)
+	podSpecificFieldsSet := make(fields.Set, 10)
 	podSpecificFieldsSet["spec.nodeName"] = pod.Spec.NodeName
 	podSpecificFieldsSet["spec.restartPolicy"] = string(pod.Spec.RestartPolicy)
 	podSpecificFieldsSet["spec.schedulerName"] = string(pod.Spec.SchedulerName)
 	podSpecificFieldsSet["spec.serviceAccountName"] = string(pod.Spec.ServiceAccountName)
 	podSpecificFieldsSet["status.phase"] = string(pod.Status.Phase)
-	// TODO: add podIPs as a downward API value(s) with proper format
 	podIP := ""
+	var podIPs []string
 	if len(pod.Status.PodIPs) > 0 {
 		podIP = string(pod.Status.PodIPs[0].IP)
+		for _, v := range pod.Status.PodIPs {
+			podIPs = append(podIPs, v.IP)
+		}
 	}
 	podSpecificFieldsSet["status.podIP"] = podIP
+	podSpecificFieldsSet["status.podIPs"] = strings.Join(podIPs, ",")
```

Comment:
It's not clear to me that a field selector on this field is desired... the use case I'm aware of for a field selector on the singular field was to be able to find the pod with a given IP address. To use this new selector you would need to know all the IPs for a pod (in the right order)... is that information typically available to API clients that don't already know the specific pod instance?

Reply:
Yes, that's why the first time in https://github.com/kubernetes/kubernetes/issues/94719 I suggest to disable `podIPs` field-selector.
Maybe we just can't support array in field-selector.